pkg/quota/quota.go

```diff
@@ -79,6 +79,14 @@ func Check(quotas []Quota, checks []Constraint) ([]ConstraintReport, error) {
 			continue
 		}
 
+		// -1 means unlimited in OpenStack
+		if matched.Limit == -1 {
```

Comment:
Let's not carry forward the smell of using -1 to mean unlimited into code that we control. Instead, add a field to `Quota` that will indicate whether the quota is unlimited.

Reply:
yeah I thought about it and wasn't sure what direction would be the best. I'll do that instead, thanks for your feedback!